src/core/sprite.js

```diff
@@ -64,17 +64,15 @@ export default class Sprite extends GameObject {
 
   //#region Static Methods
 
-  static restore(data) {
+  static async restore(data) {
```

Comment:
Not sure if restore has any benefit on being async on classes that use ContentLoader, since it's always invoked synchronously. Maybe try to change that?